apoc.merge.node escapes twice the identProp

### Versions

  - OS: irrelevant
  - Neo4j: 4.1.5
  - Neo4j-Apoc: 4.1.5


The following query fails since APOC is escaping twice the identProp : 

```
CALL apoc.merge.node(["test"], apoc.map.fromPairs([["i:d","123"]])) YIELD node return node
```


```
Failed to invoke procedure `apoc.merge.node`: Caused by: org.neo4j.exceptions.SyntaxException: Invalid input 'i': expected '`', whitespace, ':' or '}' (line 1, column 17 (offset: 16))
"MERGE (n:test{``i:d``:$identProps.``i:d``}) ON CREATE SET n += $onCreateProps ON MATCH SET n += $onMatchProps RETURN n"
```

--- a/core/src/main/java/apoc/merge/Merge.java
+++ b/core/src/main/java/apoc/merge/Merge.java
@@ -83,7 +83,7 @@ public class Merge {
     private String buildIdentPropsString(Map<String, Object> identProps) {
         if (identProps == null) return "";
         return identProps.keySet().stream().map(Util::quote)
-                .map(s -> "`"+s+"`:$identProps.`" + s+"`")
+                .map(s -> s + ":$identProps." + s)
                 .collect(Collectors.joining(","));
     }
 }

--- a/core/src/test/java/apoc/merge/MergeTest.java
+++ b/core/src/test/java/apoc/merge/MergeTest.java
@@ -16,8 +16,6 @@ import static apoc.util.TestUtil.testCall;
 import static apoc.util.TestUtil.testResult;
 import static org.junit.Assert.*;
 
-import static apoc.util.TestUtil.testCall;
-
 public class MergeTest {
 
     @Rule

--- a/core/src/test/java/apoc/merge/MergeTest.java
+++ b/core/src/test/java/apoc/merge/MergeTest.java
@@ -16,6 +16,8 @@ import static apoc.util.TestUtil.testCall;
 import static apoc.util.TestUtil.testResult;
 import static org.junit.Assert.*;
 
+import static apoc.util.TestUtil.testCall;
+
 public class MergeTest {
 
     @Rule
@@ -94,7 +96,22 @@ public class MergeTest {
             }
         }
     }
-
+    
+    @Test
+    public void testEscapeIdentityPropertiesWithSpecialCharactersShouldWork() {
+        for (String key: new String[]{"normal", "i:d", "i-d", "i d"}) {
+            Map<String, String> identProps = MapUtil.map(key, "value");
+            Map<String, Object> params = MapUtil.map("identProps", identProps);
+            
+            testCall(db, "CALL apoc.merge.node(['Person'], $identProps) YIELD node RETURN node", params,
+                        (row) -> {
+                            assertTrue(row.get("node") instanceof Node);
+                            assertTrue(node.hasProperty(key));
+                            assertEquals("value", node.getProperty(key));
+                        });
+        }
+    }
+    
     @Test
     public void testLabelsWithSpecialCharactersShouldWork() {
         for (String label: new String[]{"Label with spaces", ":LabelWithColon", "label-with-dash", "LabelWithUmlautsÄÖÜ"}) {

